rfcs/000-ml-feature-discovery.md

```diff
@@ -0,0 +1,183 @@
+- Feature <FEATURE/FEATURE_OF> Database
+    - Availability
+- Feature <COLUMN/COLUMN_OF> Column <> Badge
+    - Partition column
+    - Note: this column key will have feature key in key instead of a table key
+- Feature <DESCRIPTION/DESCRIPTION_OF> Description
+- Feature <DESCRIPTION/DESCRIPTION_OF> Programmatic_Description
+- Feature <QUERY_OF/HAS_QUERY> Query
+    - Code snippet/SQL: relating it to query from dashboard because we need an external link too
+- Feature <STAT/STAT_OF> Stat
+- Feature <HAS_BADGE/BADGE_FOR> Badge
+    - contains PII
+- Feature <OWNER_TAG/TAGGED_BY_OWNER> Tag
+    - Owner Tags, tag type can now be “owner”, not just default
+- Feature <TAG/TAGGED_BY> Tag
+    - Amundsen User Tags
+- Feature <OWNER/OWNER_OF> User
+- Feature <WATERMARK> Watermark
+    - Date range
+
+### Amundsen Common
+Amundsen common will have the folllowing schema which represents the feature response between servies:
+```
+class Feature:
+   key: str  # ex: test_feature_group_name/test_feature_name/1.2.0
+   name: str
+   version: str  # ex: "1.2.0"
+   status: str
+   feature_group: str
+   entity: Optional[List[str]]
+   data_type: Optional[str]
+   availability: List[str]
+   description: Optional[str]
+   owners: List[User]
+   badges: List[Badge]
+   owner_tags: Optional[List[Tag]] # non editable
+   tags: List[Tag] # editable
+   generation_code: Optional[Query]  # reuse Query entity from Dashboards
+   programmatic_descriptions: List[ProgrammaticDescription] 
+   watermarks: List[Watermark]  # date range
+   stats: List[Stat]
+   last_updated_timestamp: Optional[int]
+   created_timestamp: int
+   partition_column: Optional[Column]
+ 
+class FeatureSummary:
+   key: str  # ex: test_feature_group_name/test_feature_name/1.2.0
+   name: str
+   version: str  # ex: "1.2.0"
+   availability: List[str]
+   entity: Optional[List[str]]
+   description: Optional[str]
+```
+### Amundsen Metadata
+The features API methods will basically be identical to that for tables, but the response will follow the outline of the common schema specified above.
+#### Requests
+- GET feature details
+- GET feature upstream lineage, leverage CompilerWorks lineage functionality
+- PUT/DELETE owners
+- GET/PUT description
+- PUT/DELETE tags
+- PUT/DELETE badges
+#### Other Work
+- Expand ResourceType to include Feature
+- Update base proxy interface and stub out new methods in all proxies
+- Implement functionality in neo4j proxy
+
+### Amundsen Search
+> TODO
+
+## Drawbacks
+
+This addittion requires a lot of work across the entirity of the Amundsen platform. The UI design proposed below was specifically designed to make the changes as minimally invasive as possible.
+
+## Alternatives
+
+- Create a generic entity that can be used as a feature.
+
+## Prior art
+
+- [Datahub ML models](https://github.com/linkedin/datahub/tree/master/docs/rfc/active/1812-ml_models)
+- [Feast](https://github.com/feast-dev/feast)
+
+## Unresolved questions
```

Comment:
It looks like this RFC is able to support materialized features well - there are attributes for where the feature is persisted , the query that uses it, etc. There are are other features that are calculated on the fly, just before model execution (e.g. with scikit learn / TF pipelines, custom transformations, etc.) which may be transient and I am curious how how well you see these features being supported. 

While I think many of these on-the-fly features may be unique to the model that they are associated this could also be an area where feature discovery helps to understand how these "last-mile" features are created.